src/main/java/org/reaktivity/specification/socks/internal/SocksFunctions.java

```diff
@@ -126,6 +160,18 @@ public String getPrefixName()
         }
     }
 
+    public static boolean vaildateIpv4(
+        String address)
+    {
+        return IPV4_ADDRESS_PATTERN.matcher(address).matches();
```

Comment:
Now let's eliminate the garbage due to creating new matchers.

Hoist matcher on empty string as thread local constant alongside other constants near top of class:
```
private static final ThreadLocal<Matcher> IPV4_ADDRESS_MATCHER = ThreadLocal.withInitial(() -> IPV4_ADDRESS_PATTERN.matcher(""));
```
Then in this `validateIpv4` method, use `reset(...)` method to reuse same `Matcher` instance.
```
    return IPV4_ADDRESS_MATCHER.get().reset(address).matches();
```
Then inline this method to the one place where it is being used.

Please apply same feedback to `validateIpv6`.

Reply:
done.

Reply:
Not quite done yet.

>Then inline this method to the one place where it is being used.

and

>Please apply same feedback to validateIpv6.